patches/server/0977-Add-mob-spawner-ranges-changes-and-API.patch

```diff
@@ -0,0 +1,117 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: DavidTs93 <david.ts93@gmail.com>
+Date: Fri, 7 Apr 2023 19:28:43 +0300
+Subject: [PATCH] Add mob spawner ranges changes and API
+
+
+diff --git a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+index 8d442c5a498ecf288a0cc0c54889c6e2fda849ce..dfd8f1186d01bff7169f687c5771224fbbf138be 100644
+--- a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
++++ b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+@@ -276,6 +276,14 @@ public class GlobalConfiguration extends ConfigurationPart {
+                 executor.setMaximumPoolSize(_chatExecutorMaxSize);
+             }
+         }
++
++        public DefaultMobSpawnerRange defaultMobSpawnerRange;
++
++        public class DefaultMobSpawnerRange extends ConfigurationPart {
++            public int horizontal = 4;
++            public int vertical = 1;
++        }
++
+         public int maxJoinsPerTick = 5;
+         public boolean fixEntityPositionDesync = true;
+         public boolean loadPermissionsYmlBeforePlugins = true;
+diff --git a/src/main/java/net/minecraft/world/level/BaseSpawner.java b/src/main/java/net/minecraft/world/level/BaseSpawner.java
+index 31ac0e5ca26c7bdfa9b710d0bb78d846ddf6863e..0e084a9516f6220065c8f2be175f74be1a036277 100644
+--- a/src/main/java/net/minecraft/world/level/BaseSpawner.java
++++ b/src/main/java/net/minecraft/world/level/BaseSpawner.java
+@@ -4,6 +4,8 @@ import com.mojang.logging.LogUtils;
+ import java.util.Optional;
+ import java.util.function.Function;
+ import javax.annotation.Nullable;
++
++import io.papermc.paper.configuration.GlobalConfiguration;
+ import net.minecraft.core.BlockPos;
+ import net.minecraft.core.particles.ParticleTypes;
+ import net.minecraft.core.registries.BuiltInRegistries;
+@@ -27,7 +29,7 @@ import net.minecraft.world.phys.AABB;
+ import org.slf4j.Logger;
+ 
+ public abstract class BaseSpawner {
+-
++    private static final String SPAWN_RANGE_VERTICAL_TAG = "Paper.SpawnRangeVertical";
```

Comment:
Make this just a line addition instead of a line replacement (and add a `// Paper` comment).